api/src/main/java/org/xnio/ssl/JsseSslConduitEngine.java

```diff
@@ -317,6 +318,14 @@ private int wrap(final ByteBuffer src, boolean isCloseExpected) throws IOExcepti
                     doFlush();
                 }
             } catch (IOException ignore) {}
+            Throwable cause = e.getCause();
+
+            while (cause!=null && !cause.equals(cause.getCause())){
+                if(cause instanceof CertPathBuilderException){
+                    throw new SSLException(e.getMessage()+", check for missing required intermediate / root certificates in a truststore",e);
```

Comment:
I'll let @bmaxwell / @fl4via confirm that this is the right place to add this.

In general, for log messages, it's good to add the message to a logger file for internationalization (e.g., see https://github.com/xnio/xnio/blob/3.x/api/src/main/java/org/xnio/_private/Messages.java).

Reply:
@fjuma / @fl4via / @moulalis it looks like it works in an ejb test I had , as this was the exception: 

javax.naming.CommunicationException: WFNAM00018: Failed to connect to remote host [Root exception is javax.net.ssl.SSLException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target, check for missing required intermediate / root certificates in a truststore]
...
Caused by: javax.net.ssl.SSLException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target, check for missing required intermediate / root certificates in a truststore

Reply:
@fjuma that's right, the message needs to be added to org.xnio._private.Messages

Reply:
> @fjuma that's right, the message needs to be added to org.xnio._private.Messages

Hi @fl4via and @fjuma I've updated the changes as expected.